### ISSUE TITLE
What's a data directory and postmaster.pid issue

### ISSUE THREAD

**Issue by MtnBiker** [PostgresApp/PostgresApp#395]
I'm having a Postgres.app 9.5 problem and the https://postgresapp.com/documentation/troubleshooting.html refers to `data directory` which I'm unfamiliar with. Long time macOS user. I found this searching for postmaster.pid which is identified as a problem. When I found this file it was dated 1.5 years ago, so unlikely its corrupted.

**Comment by jakob** [comment]
The data directory of your PostgreSQL server can be changed in the settings. The default data dir for version 9.5 is in `~/Library/Application Support/Postgres/var-9.5`.

postmaster.pid stores the pid (process id) of the main postgres process. If it was created 1.5 years ago you either haven't restarted PostgreSQL in a very long time, or the server was shut down ungracefully and wasn't started in a long time.

You can delete the postmaster.pid file **if you make sure that no process named `postgres` or `postmaster` is running anymore**.

**Comment by MtnBiker** [comment]
Thanks jakob. The comment had more to do with the documentation—data directory is not a term known to me as a Mac user. Application Support I do know about. Is there a better place to comment on the documentation?

Ah, the `postmaster.pid` I found was another one: `/usr/local/var/postgres/postmaster.pid,` maybe from some other attempted installation of Postgres when I was getting started with it.  In any case what got me going was deleting another file: `com.postgresapp.Postgres2LoginHelper.plist`. Found that via Stackoverflow. All due to a flaky dock for my laptop which is now awaiting an RWA.

**Comment by jakob** [comment]
Good point. It would be good to write some documentation on the data directory, since that is something that is often unclear to people new to PostgreSQL. (I remember that it was unclear to me as well).

Then we should probably link to that page any time the data directory is mentioned in the docs.

**Comment by jerry-shao** [comment]
Open Postgres GUI app, click **server settings**, you will see the data path.

By default, it may stored at `/Users/USERNAME/Library/Application Support/Postgres/var-POSTGRES_VERSION`, change `USERNAME` and `POSTGRES_VERSION` to yours.

**Comment by jcdavison** [comment]
@jerry-shao crazy when I find a year old post that solves my problems so smartly, and it was weird, my machine reset itself while I was in restroom and I came back to this prob ⚡️

**Comment by SofaKing18** [comment]
> You can delete the postmaster.pid file **if you make sure that no process named `postgres` or `postmaster` is running anymore**.

@chrispysoft what do you think about GUI button "remove postmaster.pid", that appears, when there are problems with this file? I would like to do this, cause have often problems with it

**Comment by anoobbava** [comment]
for anyone, who comes later,

```https://stackoverflow.com/questions/17800249/postgres-db-not-starting-on-mac-osx-error-says-connections-on-unix-domain-sock```

**Comment by softwaregravy** [comment]
Could I suggest this become a feature request to include the path to the file in the message so we don't have to go Google to find it?

**Comment by MtnBiker** [comment]
Another way to solve the problem is:

on command line:
 lsof -i tcp:3000
kill -9 <process with ruby from result of above>

**Comment by dangayle** [comment]
I came here from Google trying to find this out. Can it be added to the error message?

<img width="712" alt="Screen Shot 2020-05-12 at 12 48 34 PM" src="https://user-images.githubusercontent.com/129611/81738909-55bd0680-944f-11ea-9476-a435a9cfa012.png">

**Comment by mikechen66** [comment]
Since there are three different libraries in Mac, developers are usually confused with the three scenarios. However, you can find out your PostgreSQL directory with the two methods on the Mac Computer. 

1. Directory--Absolute Path 

1). Scroll up your mouse to the left top of Mac Screen 

2). Click the menu "Go"

3). Select the "Go to Folder..." 

4). Input the path 

/Users/username/Library/Application Support/Postgres

Please change "username" to your name such as "jack" listed as follows. 

/Users/jack/Library/Application Support/Postgres

5). Choose the button "Go" to the directory of Postgres. 

You will find out Postgres. 


2. Find Directory step y step 

1). Scroll up your mouse to the left top of Mac Screen. 

2). Click the menu "Go". 

3). Select the button "Library"

4). Double click the file holder "Application Support". 

5). Find out "Postgres" according to the alphabetic order.

**Comment by jakob** [comment]
Instead of posting even more solutions, please just read the comment from @jerry-shao above if you are looking for the data directory.
